src/Tokenizer.ts

```diff
@@ -0,0 +1,192 @@
+            throw new MethodParamsTokenizerException(`Expected : after parameter ${lastName}.`);
+        }
+        this.tokens.push(new MethodParamsToken(TokenType.PARAMETER_NAME_VALUE_SEPARATOR, this.raw[idx]));
+
+        idx = this._ignore(/\s/, idx + 1);
+        try {
+            idx = this.parameterValue(idx);
+        } catch (e) {
+            const paramName = this.tokens[this.tokens.length - 2].value;
+            e.message = `${e.message} in parameter ${paramName}.`;
+            throw e;
+        }
+        return this._ignore(/\s/, idx);
+    }
+
+    parameterName(idx:number):number {
+        const nameEnd = this._ignore(/\w/, idx);
+        const name = this.raw.slice(idx, nameEnd);
+        if (!name)
+            throw new MethodParamsTokenizerException(`Expected parameter name, found ${this.raw[idx]}`);
+
+        this.tokens.push(new MethodParamsToken(TokenType.PARAMETER_NAME, name));
+        return nameEnd;
+    }
+
+    parameterValue(idx:number):number {
+        if (this.raw[idx].match(/('|")/))
+            return this.stringLiteral(idx);
+
+        const valueEnd = this._until(/\s|,|\)/, idx);
+        const value = this.raw.slice(idx, valueEnd);
+        if (!this._checkPrimitiveValue(value))
+            throw new MethodParamsTokenizerException(`Invalid value ${value}`);
+
+        this.tokens.push(new MethodParamsToken(TokenType.PARAMETER_VALUE, value));
+        return valueEnd;
+    }
+
+    stringLiteral(idx:number):number {
+        const delimiter = this.raw[idx];
+        const matchStep = 2;
+        const matchedEnd = this._until(new RegExp(`([^\\\\]${delimiter})|\\n`), idx, matchStep);
+        if (this.raw.slice(matchedEnd, matchedEnd + matchStep).match(/\n/)) {
+            throw new MethodParamsTokenizerException(`Invalid multiline string literal.`);
+        }
+
+        const literalEnd = matchedEnd + matchStep;
+        if (this.raw[literalEnd - 1] !== delimiter) {
+            throw new MethodParamsTokenizerException(`Mismatched string literal delimiter ${delimiter}.`);
+        }
+        const literal = this.raw.slice(idx, literalEnd);
+        this.tokens.push(new MethodParamsToken(TokenType.PARAMETER_VALUE, literal));
+        return literalEnd;
+    }
+
+    _checkPrimitiveValue(value:string):boolean {
+        // Don't accept characters other than letters, digits, _, . and -
+        // It must have at least one letter, digit or _
+        const hasSomeChar = value.length > 0;
+        const invalidChar = !!value.match(/[^A-Z0-9_\.-]/i);
+        if (!hasSomeChar || invalidChar)
+            return false;
+        if (value.match(/[A-Z_]/i))
+            return this._checkNameValue(value);
+        return this._checkNumberValue(value);
+    }
+
+    _checkNameValue(value:string):boolean {
+        // A name must not start with digits, or have non word characters
+        return !value.match(/^\d/) && !value.match(/\W/);
+    }
+
+    _checkNumberValue(value:string):boolean {
```

Comment:
You could use parseInt for this. There are other alternatives too.

Reply:
Good idea, I had forgotten about parseInt and parseFloat. However there would still need to have some checks using them. 
`parseInt('-0')` gives -0 (maybe that's ok?),
`parseInt('123ab')` gives 123
`parseInt('-1-2')`gives -1
`parseFloat('.3')` gives 0.3
`parseFloat('1.2.3.') gives 1.2
`parseFloat('1.2.')` gives 1.2

Maybe there's some more complete utility for this?

Reply:
How about `Number()`?
Obs: this is ok: `parseFloat('.3')` gives 0.3

Reply:
Perfect!